cli/graph_util.rs

```diff
@@ -528,7 +528,9 @@ impl ModuleGraphBuilder {
     // opted into using a node_modules directory
     if self.options.node_modules_dir_enablement() == Some(true) {
       if let Some(npm_resolver) = self.npm_resolver.as_managed() {
-        npm_resolver.ensure_top_level_package_json_install().await?;
+        npm_resolver
+          .ensure_top_level_package_json_install(self.options.frozen_lockfile())
```

Comment:
Since this will remain static the entire time, maybe we should just pass this in via the constructor of ManagedCliNpmResolver?

Alternatively, perhaps it might make sense to create something like `CliLockfile` which holds the `Mutex<Lockfile>` along with whether it's frozen? Then `lockfile.write_if_has_changes()` could error when `self.frozen` (we could probably put all the methods in cli/args/mod.rs into that struct as well). It looks like it would be much nicer because it would allow us to bury the lockfile Mutex too and then we could put some helper methods for the CLI onto it (might be good to land that change in a separate PR before this one).

Reply:
Updated on top of https://github.com/denoland/deno/pull/24366